src/main/scala/AutoGen.scala

```diff
@@ -51,12 +56,16 @@ object AutoGen extends LowPriorityGenProviderInstances {
     }
   })
 
+  implicit def providerLocalDate(s: String): GenProvider[LocalDate] = instance({
+    s.toLowerCase match {
+      case "dateofbirth" | "dob" | "birthdate" | "bornon" | "birthday" ⇒ Gen.age.map(a ⇒ LocalDate.now().minusDays(a.toLong))
+      case _                                                           ⇒ Gen.date
```

Comment:
Will LocalDate.now introduce non-determinism? And should that be minusYears not minusDays?

Reply:
Yes it will - and yes I meant minusYears. If there is an age field and we always want that to return the same value then if there's a field which is the date of birth and we want consistent data then the date of birth is going to change year with time. The question is then what is the best thing to do in this case?

Reply:
actually, does it matter about non-determinism here? What do you mean by non-dterminism here? This method is used to provide a `Gen` instance for lcoal date. Here I'm using the sensible distribution of ages from `Gen.age` to derive a sensible date of birth.

Reply:
Because of the call to LocalDate.now I think the Gen will return a different result for the same seed from one day to the next. Maybe the only solution is to have an "epoch" i.e. use a hard date rather than LocalDate.now.